tests/console/runc.pm

```diff
@@ -0,0 +1,103 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test runc installation and extended usage
+#    Cover the following aspects of runc:
+#      * package can be installed
+#      * create specification files
+#      * run the container
+#      * complete lifecycle (create, start, pause, resume, kill, delete)
+# Maintainer: Panagiotis Georgiadis <pgeorgiadis@suse.com>
+
+use base "consoletest";
+use testapi;
+use utils;
+use strict;
+
+# Setup the required testing environment
+sub setup {
+    # install the docker package if it's not already installed
+    zypper_call('in docker');
+
+    # make sure docker daemon is running
+    systemctl('start docker');
+    systemctl('status docker');
+}
+
+sub create_oci_bundle {
+    # create the rootfs directory
+    assert_script_run('mkdir ./rootfs');
+
+    # export busybox via Docker into the rootfs directory
+    assert_script_run('docker export $(docker create busybox) | tar -C rootfs -xvf -');
+}
+
+sub run {
+    select_console("root-console");
+
+    # runC cannot create or extract the root filesystem on its own. Use Docker to create it.
+    record_info 'Setup', 'Setup the environment';
+    setup;
+    create_oci_bundle;
+
+    # installation of runc package
+    record_info 'Test #1', 'Test: Installation';
+    if (is_caasp && check_var('FLAVOR', 'DVD') && !check_var('SYSTEM_ROLE', 'plain')) {
+        # runC should be pre-installed in MicroOS
+        die "runC is not pre-installed." if script_run("zypper se -x --provides -i runc | grep runc");
+    }
+    else {
+        zypper_call("in runc");
+    }
+
+    # create the OCI specification and verify that the template has been created
+    record_info 'Test #2', 'Test: OCI Specification';
+    assert_script_run('runc spec');
+    assert_script_run('ls -l config.json');
+    script_run('cp config.json config.json.backup');
+
+    # Modify the configuration to run the container in background
+    assert_script_run("sed -i -e '/\"terminal\":/ s/: .*/: false,/' config.json");
+    assert_script_run("sed -i -e 's/\"sh\"/\"echo\", \"Kalimera\"/' config.json");
+
+    # Run (create, start, and delete) the container after it exits
+    record_info 'Test #3', 'Test: Use the run command';
+    validate_script_output('runc run test1', sub { m/Kalimera/ });
```

Comment:
It's up to your preference, but using grep is faster and maybe more readable - I think validate_script_output was meant for things like https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/puppet.pm#L34

I would use this instead:
assert_script_run("runc run test1 | grep Kalimera")

Reply:
good point ;) Let me try that